File: leetcode/python/1758_minimum_changes_to_make_alternating_binary_string.py
class Solution:
    def minOperations(self, s: str) -> int:
        start0: int = 0
        for i, b in enumerate(s):
            if i % 2 == 0:
                if b != '0':
                    start0 += 1
            else:
                if b != '1':
                    start0 += 1
        return min(start0, len(s) - start0)

File: leetcode/rust/src/1758_minimum_changes_to_make_alternating_binary_string.rs
impl Solution {
    pub fn min_operations(s: String) -> i32 {
        let mut start0 = 0;
        // 0101
        for (i, c) in s.chars().enumerate() {
            if i % 2 == 0 && c == '1' {
                start0 += 1;
            } else if i % 2 == 1 && c == '0' {
                start0 += 1;
            }
        }
        start0.min(s.len() - start0) as i32
    }
}